### PR TITLE
Update `cortex-m-rt` version in examples to 0.7.3

### PR DESCRIPTION
**Pull request by markhakansson**
## Summary
Bumps the `cortex-m-rt` versions in the lockfiles in the examples to 0.7.3.

Due to the discovery in https://github.com/rust-embedded/cortex-m/discussions/469, it is not advisable to be using 0.7.1 or 0.7.2 of `cortex-m-rt` anymore.

### CODE REVIEW

**Comment by markhakansson** [comment]
Also, would it be a good idea to also explicitly use the 0.7.3 version of `cortex-m` in all PACs? Or is that responsibility better placed in the hands of the users of this crate?

**Comment by michalfita** [comment]
No, I wouldn't bet on that. It's better to explicitly set the minimum.

I'll try to address this this week, unless someone else throws PR. It would require a PACs release.

**Comment by michalfita** [comment]
@markhakansson the general problem is now addressed in new PACs that are subject to release. But I'd like this change in `examples` as well. Could you, please, update the `CHANGELOG.md` appropriately for CI to pass? I can't push to a branch on your fork w/o your explicit permission (which I believe is GH flaw, as maintainer of the target repo I should be allowed to push fixes to PRs before merging).

**Comment by markhakansson** [comment]
> Could you, please, update the CHANGELOG.md appropriately for CI to pass?

Sure thing! I'll fix it right away.

>  I can't push to a branch on your fork w/o your explicit permission (which I believe is GH flaw, as maintainer of the target repo I should be allowed to push fixes to PRs before merging).

Interesting, as long as  `Allow edits by maintainers` is enabled (as it is here) on the PR I would assume that it should work.

**Comment by markhakansson** [comment]
@michalfita I rebased and added the changelog entry now.

**Comment by michalfita** [comment]
> Interesting, as long as Allow edits by maintainers is enabled (as it is here) on the PR I would assume that it should work.

To be honest I haven't tried with this one in particular, but my experience shows that's mostly the case. It rings a bell such option exists, but I think most people doesn't enable it.

Thanks for the effort anyway.

**Review by michalfita — APPROVED** [review]
We want this change in as it's in alignment with new PACs minimum version of `cortex-m-rt`.

**Pull request merged**
